src/playercards/AllCardsBag.ttslua

```diff
@@ -27,20 +184,49 @@ end
 function clearIndexes()
   indexingDone = false
   cardIdIndex = { }
-  classAndLevelIndex = { }
-  classAndLevelIndex["Guardian-upgrade"] = { }
-  classAndLevelIndex["Seeker-upgrade"] = { }
-  classAndLevelIndex["Mystic-upgrade"] = { }
-  classAndLevelIndex["Survivor-upgrade"] = { }
-  classAndLevelIndex["Rogue-upgrade"] = { }
-  classAndLevelIndex["Neutral-upgrade"] = { }
-  classAndLevelIndex["Guardian-level0"] = { }
-  classAndLevelIndex["Seeker-level0"] = { }
-  classAndLevelIndex["Mystic-level0"] = { }
-  classAndLevelIndex["Survivor-level0"] = { }
-  classAndLevelIndex["Rogue-level0"] = { }
-  classAndLevelIndex["Neutral-level0"] = { }
+  allCardsIndex = { }
+  otherIndex = { }
+  classIndex = { }
+  classIndex["guardian"] = { }
+  classIndex["seeker"] = { }
+  classIndex["rogue"] = { }
+  classIndex["mystic"] = { }
+  classIndex["survivor"] = { }
+  classIndex["neutral"] = { }
+  classIndex[1] = { }
+  classIndex[2] = { }
+  classIndex[3] = { }
+  levelIndex = { }
+  levelIndex[0] = { }
+  levelIndex[1] = { }
+  levelIndex[2] = { }
+  levelIndex[3] = { }
+  levelIndex[4] = { }
+  levelIndex[5] = { }
   cycleIndex = { }
+  traitsIndex = { }
+  willpowerIconsIndex = { [0] = { } }
+  intellectIconsIndex = { [0] = { } }
+  combatIconsIndex    = { [0] = { } }
+  agilityIconsIndex   = { [0] = { } }
+  wildIconsIndex      = { [0] = { } }
+  typeIndex = { }
+  slotIndex = { }
+  slotIndex["arcane"] = { }
+  slotIndex["arcane x2"] = { }
+  slotIndex["ally"] = { }
+  slotIndex["accessory"] = { }
+  slotIndex["body"] = { }
+  slotIndex["tarot"] = { }
+  slotIndex["hand"] = { }
+  slotIndex["hand x2"] = { }
```

Comment:
For easier maintainability in the future, how about not specifying the subtables of the slotIndex and instead initializing them when the first card with that slot is encountered?
I doubt that we'll get a new slot in the near future, but maybe we get more Masks or Footwears and then it would be neat to be able to search by that if we update the metadata accordingly?

Reply:
I've got no problem with this inherently (we already do that for things like cost), but it would mean adding yet another extra check to the already lengthy loop.  I doubt it would have a noticeable impact, but still seems worth mentioning.

If we still think that's fine, happy to make that change

Reply:
Good point. I agree that we should prioritize performance in this case even if the difference won't be big.